src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -1660,7 +1580,8 @@ async def delete(self, attachments):
 
         await self._service_change('iscsitarget', 'reload')
 
-        for lun_id in lun_ids:
+        # For SCALE, reload action will remove existing LUN(s)
+        for lun_id in (lun_ids if osc.IS_FREEBSD else ()):
             await run(['ctladm', 'remove', '-b', 'block', '-l', str(lun_id)], check=False)
```

Comment:
I has been discussed with mav that this should not be necessary, right @themylogin ?

Shall we remove this block

Reply:
@william-gr we still need this block in order to wait for LUN(s) to be removed

Reply:
@themylogin that already happens on reload, so this block is actually prone to race condition so I dotn see the point in keeping it here.

We should add a sleep after reload at least until there are events from OS.

Reply:
@sonicaj please replace this with `sleep(10)` everywhere except for `terminate_luns_for_pool`